src/test/christophermanahan/github/io/carnitas/SocketInputTest.java

```diff
@@ -0,0 +1,44 @@
+package christophermanahan.github.io.carnitas;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.Socket;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class SocketInputTest {
+
+  @Test
+  void isEmptyIfSocketRespondsWithoutData() {
+    byte[] testData = new byte[0];
+    Socket testSocket = new TestSocket(testData);
+
+    assertTrue(new SocketInput(testSocket).receive().isEmpty());
+  }
+
+  @Test
+  void containsDataIfSocketRespondsWithData() {
+    String testString = "data";
+    byte[] testData = testString.getBytes();
+    Socket testSocket = new TestSocket(testData);
```

Comment:
FWIW, we don't have to extract temporary variables for each operation. This might be a little more succinct with 

```java
String testString = "data";
Socket testSocket = new TestSocket(testString.getBytes());
```

Reply:
Further, `TestSocket` could just take a string since the byte array is not being used anywhere else in the test.